### PR TITLE
Add InMemoryDocumentStore tests

### PR DESCRIPTION
**Pull request by belemaire**

### CODE REVIEW

File: ern-cauldron-api/src/InMemoryDocumentStore.js [hunk 1]
```diff
@@ -12,10 +12,29 @@ export default class InMemoryDocumentStore implements ICauldronDocumentStore {
   _isPendingTransaction: boolean
 
   constructor (cauldron: Object) {
+    if (!cauldron) {
+      throw new Error(`A Cauldron document object must be provided`)
```
Comment (krunalsshah) [inline ern-cauldron-api/src/InMemoryDocumentStore.js]: minor : use ' for `

File: ern-cauldron-api/src/InMemoryDocumentStore.js [hunk 2]
```diff
@@ -12,10 +12,29 @@ export default class InMemoryDocumentStore implements ICauldronDocumentStore {
...
+    }
     this._pendingCauldron = cauldron
     this._isPendingTransaction = false
   }
 
+  get currentCauldron () : Object {
```
Comment (krunalsshah) [inline ern-cauldron-api/src/InMemoryDocumentStore.js]: the name of the method is not clear, does it return cauldronConfig?

File: ern-cauldron-api/src/InMemoryDocumentStore.js [hunk 3]
```diff
@@ -12,10 +12,29 @@ export default class InMemoryDocumentStore implements ICauldronDocumentStore {
...
+    return this._pendingCauldron
+  }
+
+  get latestCommitedCauldron () : Object {
```
Comment (krunalsshah) [inline ern-cauldron-api/src/InMemoryDocumentStore.js]: latestCauldron<what>?

**Comment by belemaire** [comment]
Review comments addressed

**Pull request merged**